app/src/main/java/org/mozilla/tv/firefox/fxa/FxaRepo.kt

```diff
@@ -44,17 +45,16 @@ class FxaRepo(
 ) {
 
     enum class AccountState {
-        // TODO: Are these states accurate?
         // TODO: Later, may need "failed to login": https://github.com/mozilla-mobile/android-components/issues/3712
-        AUTHENTICATED,
-        AUTHENTICATED_NO_PROFILE, // todo: necessary? profile seems to be fetched async so yes?
-        NEEDS_REAUTHENTICATION,
-        NOT_AUTHENTICATED
-        // TODO: do we need initial state before state is fetched from disk?
+        AUTHENTICATED_WITH_PROFILE, // After the profile is fetched async
+        AUTHENTICATED_NO_PROFILE, // Before the profile is fetched async
```

Comment:
I'm a little bit confused by this comment.  Is this state active while a fetch is in progress?  That seems counterintuitive, given the name.

Reply:
Ya this is the first state after logging in. It is still in authenticated state but the profile is being fetched. Then once the profile is fetched, it moves to authenticated_with_profile. Is there a different name that you think would make it more intuitive?

Reply:
I can see what Severin means: it sounds like AUTHENTICATED_NO_PROFILE and AUTHENTICATED_WITH_PROFILE are two different terminal states. The comments help but I can't think of a better name quickly. Maybe AUTHENTICATED_BEFORE_PROFILE? AUTHENTICATED_FETCHING_PROFILE?

Can you make the inline comments kdoc comments above each enum state?

Reply:
To me, fetching the profile seems to be visually the same state as having no profile. It is possible that the profile is null, and in that case this would be the state. Unless we want to do something visually different while it is fetching, makes sense for me to keep them in the same state.

Reply:
Seems reasonable. :)